### PR TITLE
B-21977 INT - Services Counselor can select counseling office when creating a move

### PR DESCRIPTION
**Pull request by ajlusk**
## Summary

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office app
2. Login as a services counselor
3. Navigate to the customer search page
4. Search for any customer, maybe Spacemen, Leo
5. Create a new move for you selected user
6. When creating orders for your new move ensure the counseling office dropdown 
7. Select a counseling office and complete the orders form (for ease make sure it's origin in the same GBLOC as the services counselor you're using)
8. Add any shipment to your newly created move so it will show up on your queue
9. Navigate to your counseling queue, find the move you just created, and ensure the list counseling office matches the office you selected in step 7. 

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

**Comment by r-mettler** [comment]
See the counseling office selection in the history log

image

Able to select counseling office as SC when creating move/shipment.

image

Don't see the counseling office dropdown when selecting GHC counseling

image

File: src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx [hunk 1]
```diff
@@ -208,3 +243,34 @@ describe('AddOrdersForm - Edge Cases and Additional Scenarios', () => {
     expect(screen.getByLabelText('Orders type')).toBeDisabled();
   });
 });
+
+describe('AddOrdersForm - With Counseling Office', () => {
+  it('displays the counseling office dropdown', async () => {
```
Comment (r-mettler) [inline src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx:248]: Can you add a test for not displaying the dropdown when GHC counseling?

**Review by r-mettler — DISMISSED** [review]

File: src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx (hunk 1 above)
Comment (ajlusk) [reply src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx:251]: Added a check for this in the preexisting test case both in this file and in ServicesCounselingAddOrders.test.jsx.

**Review by r-mettler — APPROVED** [review]
New test updates look good.

File: src/components/Office/AddOrdersForm/AddOrdersForm.jsx [hunk 2]
```diff
@@ -140,10 +161,29 @@ const AddOrdersForm = ({
...
                   setCurrentDutyLocation(e);
+                  handleCounselingOfficeChange();
                 }}
                 metaOverride={originMeta}
                 required
               />
+              {currentDutyLocation.provides_services_counseling && (
+                <div>
+                  <Label>
+                    Select an origin duty location that most closely represents your current physical location, not
```
Comment (ryan-mchugh) [inline src/components/Office/AddOrdersForm/AddOrdersForm.jsx:172]: AC puts this text into third person, referencing the customer instead of acting as the customer.

"Select an origin duty location that most closely represents the customers current physical location, not where their shipment will originate, if different. This will allow a nearby transportation office to assist them."

File: src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx [hunk 3]
```diff
@@ -208,3 +246,33 @@ describe('AddOrdersForm - Edge Cases and Additional Scenarios', () => {
...
+    const selectedOptionCurrent = await screen.findByText(/Scott/);
+    await userEvent.click(selectedOptionCurrent);
+
+    // Test New Duty Location Search Box interaction
+    await userEvent.type(screen.getByLabelText(/New duty location/), 'AFB', { delay: 100 });
+    const selectedOptionNew = await screen.findByText(/Luke/);
+    await userEvent.click(selectedOptionNew);
+
+    const counselingOfficeLabel = await screen.queryByText(/Counseling office/);
+    expect(counselingOfficeLabel).toBeTruthy();
```
Comment (ryan-mchugh) [inline src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx:276]: Can you add a check to also select a counseling office in its dropdown?

File: src/pages/Office/ServicesCounselingAddOrders/ServicesCounselingAddOrders.test.jsx [hunk 4]
```diff
@@ -370,6 +388,32 @@ describe('ServicesCounselingAddOrders component', () => {
...
+
+    // Test Current Duty Location Search Box interaction
+    await user.type(screen.getByLabelText('Current duty location'), 'AFB', { delay: 500 });
+    const selectedOptionCurrent = await screen.findByText(/Hill/);
+    await user.click(selectedOptionCurrent);
+
+    const counselingOfficeLabel = await screen.queryByText(/Counseling office/);
+    expect(counselingOfficeLabel).toBeTruthy();
+
+    await user.type(screen.getByLabelText('New duty location'), 'AFB', { delay: 500 });
```
Comment (ryan-mchugh) [inline src/pages/Office/ServicesCounselingAddOrders/ServicesCounselingAddOrders.test.jsx:412]: same here, need to test that an one of the options is available in the dropdown for counseling office

File: pkg/handlers/ghcapi/transportation_offices_test.go [hunk 5]
```diff
@@ -145,3 +148,57 @@ func (suite *HandlerSuite) TestGetTransportationOfficesGBLOCsHandler() {
...
+		DutyLocationID: *handlers.FmtUUID(origDutyLocation.ID),
+	}
+
+	handler := ShowCounselingOfficesHandler{
+		HandlerConfig:                suite.HandlerConfig(),
+		TransportationOfficesFetcher: fetcher}
+
+	response := handler.Handle(params)
+	suite.Assertions.IsType(&transportationofficeop.ShowCounselingOfficesOK{}, response)
+	responsePayload := response.(*transportationofficeop.ShowCounselingOfficesOK)
```
Comment (ryan-mchugh) [inline pkg/handlers/ghcapi/transportation_offices_test.go:199]: Could finish this test off with a check that the payload matches what you sent in, so returned counseling office matches transportation office "PPPO Travis AFB -USAF"
